components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/softwareversionchecktrigger.go

```diff
@@ -1,16 +1,98 @@
 package trigger
 
-import "github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
 
-type SoftwareVersionCheck struct{}
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/fiberutils"
+	"github.com/sirupsen/logrus"
+)
+
+type SWVersionCheck interface {
+	Run(config models.Config) map[string]models.CheckTriggerResponse
+}
+
+type SoftwareVersionCheck struct {
+	log logrus.Logger
+}
 
 func NewSoftwareVersionCheck() *SoftwareVersionCheck {
-	return &SoftwareVersionCheck{}
+	return &SoftwareVersionCheck{
+		log: *logrus.New(),
+	}
 }
 
 func (ss *SoftwareVersionCheck) Run(config models.Config) map[string]models.CheckTriggerResponse {
-	m := map[string]models.CheckTriggerResponse{
-		"f": models.CheckTriggerResponse{},
+	return runCheck(config, constants.SoftwareVersions)
+}
+
+func runCheck(config models.Config, path string) map[string]models.CheckTriggerResponse {
```

Comment:
if you want to use this as a common method for all triggers, then you need to place this in another file.